app/services/case_linking_service.rb

```diff
@@ -11,37 +14,26 @@ def initialize(user, kase, link_case_number)
 
   def create
     ActiveRecord::Base.transaction do
+      begin
 
-      # By using a LinkedCase object here we can retrieve validation errors
-      # from the models
-      @case_link = LinkedCase.new(
-        linked_case_number: @link_case_number.to_s&.strip
-      )
-
-      #Create the links
-      @case.related_case_links << @case_link
-
-      if @case.valid?
-        @linked_case = @case_link.linked_case
-        # Log event in both cases
-        @case.state_machine.link_a_case!(acting_user: @user,
-                                         acting_team: @user.teams_for_case(@case).first,
-                                         linked_case_id: @linked_case.id)
-        @linked_case.state_machine.link_a_case!(acting_user: @user,
-                                                acting_team: @user.teams_for_case(@case).first,
-                                                linked_case_id: @case.id)
-        @result = :ok
-      else
-        add_case_link_errors_to_case(@case_link, @case)
-        @result = :validation_error
+        # By using a LinkedCase object here we can retrieve validation errors
+        # from the models
+        @case_link = LinkedCase.new(
+            linked_case_number: @link_case_number.to_s&.strip
+        )
+        create_links
+        @result
+      rescue CaseLinkingError => err
+        @case.errors.add(:linked_case_number, err.message)
+        @error = err
+        @result = :error
+      rescue => err
```

Comment:
Can we let Sentry handle these so that we actually see the error? Otherwise it'll get lost in the logs if this ever happens in production.